lib/streamlit/runtime/runtime.py

```diff
@@ -140,6 +141,25 @@ class AsyncObjects(NamedTuple):
     stopped: asyncio.Future[None]
 
 
+async def _cancel_async_task(task: asyncio.Task) -> None:
+    """Cancel a task and wait for it to finish.
+
+    Implementation inspired by:
+    https://github.com/python/cpython/issues/103486
+    """
+
+    task.cancel()
+    try:
+        await task
```

Comment:
I'm not sure if this is necessary in our case. We just want to have this task canceled, but we don't really care when it happens.

Reply:
I think `task.cancel()` is enough for our use-case. I will clean this up for now.